src/anemoi/inference/inputs/ekd.py

```diff
@@ -413,7 +413,14 @@ def _create_input_state(
         State
             The created input state.
         """
+        if date is None:
```

Comment:
If `date` can be `None`, the type hint should also be changed to `Optional[Date]`.

Reply:
Yes! Fixed.